### ISSUE TITLE
Incorrect types exported for Page component

### ISSUE THREAD

**Issue by BreakBB** [wojtekmaj/react-pdf#1542]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [x] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

It looks like the type definitions shipped with `react-pdf` are not correctly exported. All props of `Page` are noted as required, but they should not.

The npm page for [@types/react-pdf](https://www.npmjs.com/package/@types/react-pdf) also states that these are deprecated, hence the issue here.

### Steps to reproduce

```ts
import {Document, Page} from "react-pdf";

const A = () => (
    <Document file="test.pdf">
        <Page pageNumber={1} />
    </Document>
);
```

A warning is risen, that required attributes are missing:

<img width="509" alt="image" src="https://github.com/wojtekmaj/react-pdf/assets/33514570/715bfb99-b9bc-4153-9162-a8f29ab88dd0">

### Expected behavior

The `Page.d.ts` file used in the `index.d.ts` should correctly mark all (?) `Page` props as optional.

It looks like the types in the`Page.tsx` are correct though 🤔 

### Actual behavior

`Page` imported directly from "react-pdf" notes all props as required.

### Additional information

As a side note: It seems strange that `react-pdf` is shipped with the `src` folder containing spec files and such, while there is a `dist` folder 😅 

### Environment

- **Browser (if applicable)**: -
- **React-PDF version**: 7.1.2
- **React version**: 18.2.0
- **Webpack version (if applicable)**: -

**Comment by wojtekmaj** [comment]
I can't reproduce this in my environment (obviously 😅). Can you share repro please?

**Comment by BreakBB** [comment]
For context, I use IntelliJ as IDE, not sure if this is an IntelliJ only issue and I have no clue how to share this as repro.

Still I looked into this a bit deeper. This is part of the shipped `Page.d.ts`:

```js
declare namespace Page {
    var propTypes: {
        canvasBackground: PropTypes.Requireable<string>;
        canvasRef: PropTypes.Requireable<NonNullable<((...args: any[]) => any) | Required<PropTypes.InferProps<{
            current: PropTypes.Requireable<any>;
        }>> | null | undefined>>;
        children: PropTypes.Requireable<PropTypes.ReactNodeLike>;
        className: PropTypes.Requireable<NonNullable<string | (string | null | undefined)[] | null | undefined>>;
        customTextRenderer: PropTypes.Requireable<(...args: any[]) => any>;
        devicePixelRatio: PropTypes.Requireable<number>;
        error: PropTypes.Requireable<NonNullable<PropTypes.ReactNodeLike | ((...args: any[]) => any)>>;
        height: PropTypes.Requireable<number>;
        imageResourcesPath: PropTypes.Requireable<string>;
        inputRef: PropTypes.Requireable<NonNullable<((...args: any[]) => any) | Required<PropTypes.InferProps<{
            current: PropTypes.Requireable<any>;
        }>> | null | undefined>>;
        loading: PropTypes.Requireable<NonNullable<PropTypes.ReactNodeLike | ((...args: any[]) => any)>>;
        noData: PropTypes.Requireable<NonNullable<PropTypes.ReactNodeLike | ((...args: any[]) => any)>>;
        onGetTextError: PropTypes.Requireable<(...args: any[]) => any>;
        onGetTextSuccess: PropTypes.Requireable<(...args: any[]) => any>;
        onLoadError: PropTypes.Requireable<(...args: any[]) => any>;
        onLoadSuccess: PropTypes.Requireable<(...args: any[]) => any>;
        onRenderError: PropTypes.Requireable<(...args: any[]) => any>;
        onRenderSuccess: PropTypes.Requireable<(...args: any[]) => any>;
        onRenderTextLayerError: PropTypes.Requireable<(...args: any[]) => any>;
        onRenderTextLayerSuccess: PropTypes.Requireable<(...args: any[]) => any>;
        pageIndex: PropTypes.Validator<number>;
        pageNumber: PropTypes.Validator<number>;
        pdf: PropTypes.Requireable<NonNullable<boolean | PropTypes.InferProps<{
            getDestination: PropTypes.Validator<(...args: any[]) => any>;
            getOutline: PropTypes.Validator<(...args: any[]) => any>;
            getPage: PropTypes.Validator<(...args: any[]) => any>;
            numPages: PropTypes.Validator<number>;
        }> | null | undefined>>;
        renderAnnotationLayer: PropTypes.Requireable<boolean>;
        renderForms: PropTypes.Requireable<boolean>;
        renderMode: PropTypes.Requireable<string>;
        renderTextLayer: PropTypes.Requireable<boolean>;
        rotate: PropTypes.Requireable<number>;
        scale: PropTypes.Requireable<number>;
        width: PropTypes.Requireable<number>;
        onCopy?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onCut?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPaste?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onCompositionEnd?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onCompositionStart?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onCompositionUpdate?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onFocus?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onBlur?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onChange?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onInput?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onReset?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onSubmit?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onInvalid?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onLoad?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onError?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onKeyDown?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onKeyPress?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onKeyUp?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onAbort?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onCanPlay?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onCanPlayThrough?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDurationChange?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onEmptied?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onEncrypted?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onEnded?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onLoadedData?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onLoadedMetadata?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onLoadStart?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPause?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPlay?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPlaying?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onProgress?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onRateChange?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onSeeked?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onSeeking?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onStalled?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onSuspend?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onTimeUpdate?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onVolumeChange?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onWaiting?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onClick?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onContextMenu?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDoubleClick?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDrag?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDragEnd?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDragEnter?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDragExit?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDragLeave?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDragOver?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDragStart?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onDrop?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onMouseDown?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onMouseEnter?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onMouseLeave?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onMouseMove?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onMouseOut?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onMouseOver?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onMouseUp?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onSelect?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onTouchCancel?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onTouchEnd?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onTouchMove?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onTouchStart?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerDown?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerMove?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerUp?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerCancel?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerEnter?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerLeave?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerOver?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onPointerOut?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onGotPointerCapture?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onLostPointerCapture?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onScroll?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onWheel?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onAnimationStart?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onAnimationEnd?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onAnimationIteration?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onTransitionEnd?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
        onToggle?: PropTypes.Requireable<(...args: any[]) => any> | undefined;
    };
}
```

The only props marked as optional (`?:`) are the ones below `width` which actually are the `EventProps` added to the `Page` component. Even though, if I understand PropTypes correctly `PropTypes.Requireable` are a different way of marking props as optional 🤔 

The warning window I shared in the first post also warn about the props which are not declared with `?`.

**Comment by wojtekmaj** [comment]
Hmmm... I actually think now that we ship our own types, types generated from propTypes should not be added to .d.ts file. However, I would like to keep propTypes for these folks who are using pure JavaScript. Not sure how to proceed with that...

**Comment by BreakBB** [comment]
I just verified this issue is indeed fixed by the linked commit and already released in v7.3.0 🥳 

Thanks @wojtekmaj 👍🏻
